shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -1024,6 +1093,18 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool, val ne
       case pat if ctx.inPattern =>
         err(msg"Unsupported pattern shape${
           if (dbg) " ("+pat.getClass.toString+")" else ""}:", pat.toLoc)(raise)
+      case Lam(pat, body) if ctx.inQuote =>
```

Comment:
There's no reason for duplicating this code (and changing the way `doGenLambdas` is interpreted when in a quote for no reason). There are only a few lines of additional stuff going on here in the middle. Just add these lines to the two `Lam` cases below.

Reply:
BTW what's the `midCtx` in the `case Lam(pat, body) if doGenLambdas`? Is it used?

Reply:
It's indeed an old unused binding that should have been removed. Feel free to remove it!